File: crates/normalize-facts/tests/fixtures/python/add_numbers/project/math_utils.py
"""Simple math utilities."""


def add(a: int, b: int) -> int:
    """Return the sum of a and b."""
    return a + b


def multiply(a: int, b: int) -> int:
    """Return the product of a and b."""
    return a * b


class Calculator:
    """Stateful calculator that records history."""

    def __init__(self) -> None:
        self.history: list[int] = []

    def compute(self, op: str, a: int, b: int) -> int:
        """Apply op ('add' or 'mul') to a and b."""
        if op == "add":
            result = add(a, b)
        else:
            result = multiply(a, b)
        self.history.append(result)
        return result

File: crates/normalize-facts/tests/fixtures/zig/add_numbers/project/math_utils.zig
const std = @import("std");

pub fn add(a: i32, b: i32) i32 {
    return a + b;
}

pub fn multiply(a: i32, b: i32) i32 {
    return a * b;
}

pub const Calculator = struct {
    result: i32,

    pub fn init() Calculator {
        return Calculator{ .result = 0 };
    }

    pub fn compute(self: *Calculator, op: []const u8, a: i32, b: i32) i32 {
        if (std.mem.eql(u8, op, "add")) {
            self.result = add(a, b);
        } else {
            self.result = multiply(a, b);
        }
        return self.result;
    }

    pub fn lastResult(self: *const Calculator) i32 {
        return self.result;
    }
};
